include/zephyr/devicetree/gpio.h

```diff
@@ -100,7 +100,7 @@ extern "C" {
  * @see DT_PHANDLE_BY_IDX()
  */
 #define DT_GPIO_LABEL_BY_IDX(node_id, gpio_pha, idx) \
```

Comment:
Please add `@deprecated Use XXX instead` in Doxygen so that they appear in https://docs.zephyrproject.org/latest/doxygen/html/deprecated.html